crates/uv-cli/src/lib.rs

```diff
@@ -1887,6 +1887,11 @@ pub struct RunArgs {
     #[arg(long, value_parser = parse_maybe_file_path)]
     pub with_requirements: Vec<Maybe<PathBuf>>,
 
+    /// Run the tool in an isolated virtual environment, rather than leveraging the base environment
+    /// for the current project, to enforce strict isolation between dependencies.
+    #[arg(long)]
+    pub isolate: bool,
```

Comment:
Explaining the difference between `--isolate` and `--no-workspace` is fairly nuanced, though I believe that both are necessary, they serve very different purposes.